src/diff/nodeChanges.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as AmfGraphTypes from "./amfGraphTypes";
+import { CategorizedChange } from "./categorizedChange";
+import { RuleCategory } from "./ruleSet";
+
+/**
+ * Class to hold differences of a JSON node
+ */
+export class NodeChanges {
+  public added?: { [key: string]: unknown } | AmfGraphTypes.Node;
+  public removed?: { [key: string]: unknown } | AmfGraphTypes.Node;
```

Comment:
added and removed don't need to have id and type but AmfGraphTypes.Node must have those. Also I want to keep both types (added/removed & Node) separate because they are different

Reply:
```suggestion
  public added: { [key: string]: unknown };
  public removed: { [key: string]: unknown };
```
I'm confused why you are including `Node`, if you want to keep the separate. And because you're excluding every property defined on `Node` in these type definitions, so there doesn't seem to be a need for it.

Reply:
if the a new node is added or an existing node is removed then I assign the node which is of type AmfGraphTypes.Node to added/removed

Reply:
I'm not sure I fully understand. Are you sating `nodeChanges.added` could be either of type `Node` or it could be an object with unknown/arbitrary properties?

Reply:
yes. I can exclude id and type as these are duplicated